### PR TITLE
vapoursynth: fix darwin build

### PR DESCRIPTION
**Pull request by wegank**
For the plugin interface, as there is no equivalent to `patchelf --add-needed` on Darwin, this PR takes the easier way back to rebuilding `vapoursynth` with `--with-plugindir=${pluginsEnv}/lib/vapoursynth` instead.

Tested `vapoursynth.withPlugins [ ffms vapoursynth-mvtools ];` on aarch64-darwin following the instructions in https://gist.github.com/phiresky/4bfcfbbd05b3c2ed8645, with #359262 and the following changes applied:

- In `motioninterpolation.vpy`, replace `vapoursynth.get_core()` with `vapoursynth.core`;
- In the command, replace `--y4m` with `-c y4m`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359263`

---
### `aarch64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>vapoursynth-editor</li>
    <li>vapoursynth-nnedi3</li>
    <li>vapoursynth-znedi3</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>av1an</li>
    <li>av1an-unwrapped</li>
    <li>python311Packages.vapoursynth</li>
    <li>python311Packages.vapoursynth.dist</li>
    <li>python312Packages.vapoursynth</li>
    <li>python312Packages.vapoursynth.dist</li>
    <li>vapoursynth</li>
    <li>vapoursynth-mvtools</li>
  </ul>
</details>

**Comment by snaakey** [comment]
#359027

**Comment by al3xtjames** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359263`

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>av1an</li>
    <li>av1an-unwrapped</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>python311Packages.vapoursynth</li>
    <li>python311Packages.vapoursynth.dist</li>
    <li>python312Packages.vapoursynth</li>
    <li>python312Packages.vapoursynth.dist</li>
    <li>vapoursynth</li>
    <li>vapoursynth-bestsource</li>
    <li>vapoursynth-bestsource.dev</li>
    <li>vapoursynth-editor</li>
    <li>vapoursynth-eedi3</li>
    <li>vapoursynth-mvtools</li>
    <li>vapoursynth-nnedi3</li>
    <li>vapoursynth-nnedi3cl</li>
    <li>vapoursynth-znedi3</li>
  </ul>
</details>

**Review by al3xtjames — APPROVED** [review]

File: pkgs/by-name/va/vapoursynth-znedi3/package.nix [hunk 1]
```diff
@@ -34,7 +34,7 @@ stdenv.mkDerivation {
   installPhase = ''
     runHook preInstall
 
-    install -D -t $out/lib/vapoursynth vsznedi3${stdenv.hostPlatform.extensions.sharedLibrary}
```
Comment (rnhmjoj) [inline pkgs/by-name/va/vapoursynth-znedi3/package.nix:37]: Is it `.so` instead of `.dylib` even on darwin?

**Review by rnhmjoj — APPROVED** [review]
I know zero about darwin, but I tested loading some plugin on linux-x86_64 and it looks ok.

File: pkgs/by-name/va/vapoursynth-znedi3/package.nix (hunk 1 above)
Comment (al3xtjames) [reply pkgs/by-name/va/vapoursynth-znedi3/package.nix:37]: The file extension is hardcoded in the Makefile: https://github.com/sekrit-twc/znedi3/blob/master/Makefile#L90

**Pull request merged**